lib-api/src/main/java/org/ergoplatform/appkit/babelfee/BabelFeeBoxState.java

```diff
@@ -0,0 +1,144 @@
+        this.pricePerToken = pricePerToken;
+        this.tokenId = tokenId;
+        this.boxCreator = boxCreator;
+        this.value = value;
+        this.tokenAmount = tokenAmount;
+    }
+
+    /**
+     * @return price offered per raw token amount
+     */
+    public long getPricePerToken() {
+        return pricePerToken;
+    }
+
+    /**
+     * @return token id this babel fee box is offering change for
+     */
+    public ErgoId getTokenId() {
+        return tokenId;
+    }
+
+    /**
+     * @return box creator or owner of the babel fee box
+     */
+    public SigmaProp getBoxCreator() {
+        return boxCreator;
+    }
+
+    /**
+     * @return overall ERG value in the box. not all is available to change for token as a small
+     * amount must remain in the box
+     */
+    public long getValue() {
+        return value;
+    }
+
+    /**
+     * @return overall ERG value available to change for token
+     */
+    public long getValueAvailableToBuy() {
+        return value - Parameters.MinChangeValue;
+    }
+
+    /**
+     * @return max token amount possible to swap at best price
+     */
+    public long maxAmountToBuy() {
+        return getValueAvailableToBuy() / pricePerToken;
+    }
+
+    /**
+     * @param nanoergs amount we want to receive, usually to pay transaction fees
+     * @return the amount of tokens to sell to receive at least his amount of nanoergs
+     */
+    public long tokensToSellForErgAmount(long nanoergs) {
+        long floorAmount = nanoergs / pricePerToken;
+        return (floorAmount * pricePerToken >= nanoergs) ? floorAmount : floorAmount + 1;
+    }
+
+    /**
+     * constructs a new babel fee box state based on this fee box with a certain token amount change
+     * done
+     *
+     * @param tokenAmountChange the token amount to add to the new babel fee box
+     * @return new babel fee box after swap was done
+     */
+    public BabelFeeBoxState buildSucceedingState(long tokenAmountChange) {
+        if (tokenAmountChange <= 0)
+            throw new IllegalArgumentException("tokenAmountChange must be greater than 0");
+        if (tokenAmountChange > maxAmountToBuy())
+            throw new IllegalArgumentException("tokenAmountChange must be less or equal maxAmountToBuy");
+
+        return new BabelFeeBoxState(pricePerToken, tokenId, boxCreator,
+            value - tokenAmountChange * pricePerToken,
+            tokenAmountChange + tokenAmount);
+    }
+
+    /**
+     * @return outbox representing this babel fee box
+     */
+    public OutBox buildOutbox(UnsignedTransactionBuilder txBuilder, @Nullable InputBox babelBox) {
+        OutBoxBuilder outBoxBuilder = txBuilder.outBoxBuilder()
+            .contract(new BabelFeeBoxContract().getContractForToken(tokenId, txBuilder.getNetworkType()))
+            .value(value)
+            .registers(ErgoValue.of(boxCreator), ErgoValue.of(pricePerToken), ErgoValue.of(babelBox != null ? babelBox.getId().getBytes() : new byte[0]));
```

Comment:
What empty new byte[0] means?

Reply:
In this case it is an unnecessary empty filler for the first box in a babel fee box chain which makes the code a bit more easy. Otherwise an array had to be constructed and used which is some more code.